quirinojess/caju-front-teste-1
# Release 1.0: Finalize and Integrate Features for Initial Launch

This PR aims to integrate the changes from the `develop` branch into the `main` branch as part of our upcoming release. The changes include new features, bug fixes, and performance improvements for the admissions management application developed in TypeScript/React.

**Main Changes:**

-   **Additional Features:**
    -   **Dashboard and New User Screen Redesign:** Implementation of a new visual standard following the brand colors.
    -   **Status Filter:** Implementation of a new filter with buttons that filter columns by status.
    -   **CPF Filter Functionality:** Improved filtering of admissions by CPF on the search screen and added a mask to standardize the element.
    -   **Form Validations:** Protections to ensure that requests are sent with correct data and implementation of clear messages for the user.
    -   **API Integration:** Integration via API/rest to cover filtering flows, status updates, and new registrations.

**Context and Motivation:** These changes are part of a test to implement improvements, aiming to increase efficiency, fix identified issues, and add necessary features for managing workflows. The changes were implemented and tested with a focus on user experience and best coding practices.

**Testing Instructions:**

-   **Verify Features:**
    -   **Dashboard Screen:** Navigate to the Dashboard screen and test all actions for approving, rejecting, and deleting admissions.
    -   **New User Screen:** Add new admissions and verify that all information is saved correctly.

**Added Dependencies:**

-   **Storybook:** Updated to the latest version to support new features and improvements.
-   **Axios and React Hook Form:** Updates for form validation and API/rest integration.

**Updated Dependencies:**

-   **React-router-dom:** Updated to the latest version to support new features and improvements.
-   **Styled-Components:** Updated to enhance component styling.

**Next Steps:**

-   **UX Improvements:** Develop a functional and responsive menu, add a footer with navigation buttons, and improve titles and paragraphs on the registration page.
-   **Comprehensive Testing:** Conduct thorough testing on APIs and services to validate system performance and integrity.
